packages/fx-core/src/component/driver/file/createOrUpdateJsonFile.ts

```diff
@@ -111,34 +105,34 @@ export class CreateOrUpdateJsonFileDriver implements StepDriver {
       invalidParameters.push("target");
     }
 
-    if (!args.appsettings || typeof args.appsettings !== "object") {
+    if (args.appsettings === undefined && args.content === undefined) {
+      invalidParameters.push("content");
+    } else if (args.appsettings !== undefined && args.content !== undefined) {
       invalidParameters.push("appsettings");
-    }
-
-    for (const value of Object.values(args.appsettings)) {
-      if (!value) {
+    } else if (args.appsettings) {
+      if (typeof args.appsettings !== "object") {
         invalidParameters.push("appsettings");
       }
+    } else if (args.content) {
+      if (typeof args.content !== "object") {
+        invalidParameters.push("content");
+      }
     }
 
     if (invalidParameters.length > 0) {
       throw new InvalidActionInputError(actionName, invalidParameters, helpLink);
     }
   }
 
-  private replaceProjectAppsettings(
-    projectAppsettings: Record<string, unknown>,
-    ymlAppsettings: Record<string, unknown>
+  private addOrUpdateJsonContent(
+    jsonContent: CommentJSONValue,
+    ymlJsonContent: Record<string, unknown>
   ) {
-    for (const item of Object.entries(ymlAppsettings)) {
-      if (typeof item[1] === "string") {
-        (projectAppsettings as any)[item[0]] = item[1];
-      } else if (typeof item[1] === "object") {
-        if ((projectAppsettings as any)[item[0]]) {
-          this.replaceProjectAppsettings((projectAppsettings as any)[item[0]], item[1] as any);
-        } else {
-          (projectAppsettings as any)[item[0]] = item[1];
-        }
+    for (const [key, value] of Object.entries(ymlJsonContent)) {
+      if (typeof value === "object") {
+        this.addOrUpdateJsonContent((jsonContent as any)[key], value as any);
```

Comment:
What if `(jsonContent as any)[key]` undefined?

Reply:
Add a short path to add object directly.